lib/disability_compensation/providers/document_upload/evss_supplemental_document_upload_provider.rb

```diff
@@ -47,24 +57,66 @@ def validate_upload_document(evss_claim_document)
   #
   # @return [Faraday::Response] The EVSS::DocumentsService API calls are implemented with Faraday
   def submit_upload_document(evss_claim_document, file_body)
+    log_upload_attempt
+
     client = EVSS::DocumentsService.new(@form526_submission.auth_headers)
     client.upload(file_body, evss_claim_document)
-  end
 
-  def log_upload_success(uploading_class_prefix)
-    StatsD.increment("#{uploading_class_prefix}.#{STATSD_PROVIDER_METRIC}.#{STATSD_SUCCESS_METRIC}")
+    # EVSS::DocumentsService uploads throw a EVSS::ErrorMiddleware::EVSSError if they fail
+    # If no exception is raised, log a success response
+    log_upload_success
+  rescue EVSS::ErrorMiddleware::EVSSError => e
+    # If exception is raised, log and re-raise the error
+    log_upload_failure
```

Comment:
do you need to pass e into this to log stuff from e here? or is everything in e useless or you getting it in here some other way?

Reply:
That's what I was getting at with the notes on `EVSS::ErrorMiddleware::EVSSError`, it's a custom error and I didn't do a deep dive into what information we get out of that, I can't actually find where it gets thrown at all. I can dig into it I just wasn't sure if it was worth it if we don't care as much about logging the EVSS errors in the migration if that makes sense